### PR TITLE
chore: unify all env vars used

### PR DESCRIPTION
**Pull request by samypr100**
## Summary

This PR declares and documents all environment variables that are used in one way or another in `uv`, either internally, or externally, or transitively under a common struct.

I think over time as uv has grown there's been many environment variables introduced. Its harder to know which ones exists, which ones are missing, what they're used for, or where are they used across the code. The docs only documents a handful of them, for others you'd have to dive into the code and inspect across crates to know which crates they're used on or where they're relevant.

This PR is a starting attempt to unify them, make it easier to discover which ones we have, and maybe unlock future posibilities in automating generating documentation for them.

I think we can split out into multiple structs later to better organize, but given the high influx of PR's and possibly new environment variables introduced/re-used, it would be hard to try to organize them all now into their proper namespaced struct while this is all happening given merge conflicts and/or keeping up to date.

I don't think this has any impact on performance as they all should still be inlined, although it may affect local build times on changes to the environment vars as more crates would likely need a rebuild. Lastly, some of them are declared but not used in the code, for example those in `build.rs`. I left them declared because I still think it's useful to at least have a reference.

Did I miss any? Are their initial docs cohesive?

Note, `uv-static` is a terrible name for a new crate, thoughts? Others considered `uv-vars`, `uv-consts`.

## Test Plan

Existing tests

### CODE REVIEW

File: crates/uv/tests/it/lock_scenarios.rs
Comment (samypr100) [inline crates/uv/tests/it/lock_scenarios.rs:1]: TODO: fix this as it's generated

File: crates/uv/src/version.rs
Comment (samypr100) [inline crates/uv/src/version.rs:1]: This seems to have been replaced by `crates/uv-cli/src/version.rs`?

File: crates/uv/tests/it/lock_scenarios.rs
Comment (samypr100) [reply crates/uv/tests/it/lock_scenarios.rs:1]: Fixed

**Comment by samypr100** [comment]
`uv-test-publish` failing seems to be due to permissions as it requires `id-token: write`

**Comment by konstin** [comment]
That's on my todo list, the `github.repository` check to not run this test from forks doesn't work as intended.

**Comment by samypr100** [comment]
> That's on my todo list, the `github.repository` check to not run this test from forks doesn't work as intended.

Make sense, I think we could add a `github.event.pull_request.head.repo.fork != true` check.

**Comment by zanieb** [comment]
I'm hyped for this. I'd love to generate the environment variables reference doc from it.

**Comment by konstin** [comment]
> Make sense, I think we could add a github.event.pull_request.head.repo.fork != true check.

Thanks, fixed in https://github.com/astral-sh/uv/pull/8168

**Review by zanieb — APPROVED** [review]

**Pull request merged**